diff --git a/cmake/modules/FindSamba.cmake b/cmake/modules/FindSamba.cmake
index 9240fcb..d09d3b2 100644
--- a/cmake/modules/FindSamba.cmake
+++ b/cmake/modules/FindSamba.cmake
@@ -28,23 +28,7 @@ find_library(SAMBA_LIBRARIES NAMES smbclient HINTS ${PC_SAMBA_LIBDIR})
 
 if(SAMBA_INCLUDE_DIR AND SAMBA_LIBRARIES)
    set(SAMBA_FOUND TRUE)
-   # check whether libsmbclient has smbc_set_context()
-   include(CheckSymbolExists)
-   include(MacroPushRequiredVars)
-   macro_push_required_vars()
-   set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} ${SAMBA_LIBRARIES})
-   set(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${SAMBA_INCLUDE_DIR})
-   check_symbol_exists(smbc_set_context "libsmbclient.h" SAMBA_HAVE_SMBC_SET_CONTEXT)
-   check_symbol_exists(smbc_option_set "libsmbclient.h" SAMBA_HAVE_SMBC_OPTION_SET)
-   macro_pop_required_vars()
-   # fail if smbc_set_context() was required but hasn't been found
-   if (SAMBA_REQUIRE_SMBC_SET_CONTEXT AND NOT SAMBA_HAVE_SMBC_SET_CONTEXT)
-      set(SAMBA_FOUND FALSE)
-   endif (SAMBA_REQUIRE_SMBC_SET_CONTEXT AND NOT SAMBA_HAVE_SMBC_SET_CONTEXT)
-   # fail if smbc_option_set() was required but hasn't been found
-   if (SAMBA_REQUIRE_SMBC_OPTION_SET AND NOT SAMBA_HAVE_SMBC_OPTION_SET)
-      set(SAMBA_FOUND FALSE)
-   endif (SAMBA_REQUIRE_SMBC_OPTION_SET AND NOT SAMBA_HAVE_SMBC_OPTION_SET)
+   set(SAMBA_HAVE_SMBC_SET_CONTEXT TRUE)
 else(SAMBA_INCLUDE_DIR AND SAMBA_LIBRARIES)
    set(SAMBA_FOUND FALSE)
    set(SAMBA_HAVE_SMBC_SET_CONTEXT FALSE)
