diff --git a/kdebase-kioslaves-4.14.3/kioslave/smb/kio_smb.h b/kdebase-kioslaves-4.14.3/kioslave/smb/kio_smb.h
index c2229ab..53e4145 100644
--- a/kdebase-kioslaves-4.14.3/kioslave/smb/kio_smb.h
+++ b/kdebase-kioslaves-4.14.3/kioslave/smb/kio_smb.h
@@ -210,6 +210,10 @@ protected:
 
 public:
 
+    mutable int smb_force_protocol = 0;
+    mutable const char* smb_min_proto = NULL;
+    mutable const char* smb_max_proto = NULL;
+
     //-----------------------------------------------------------------------
     // smbclient authentication callback (note that this is called by  the
     // global ::auth_smbc_get_data() call.
diff --git a/kdebase-kioslaves-4.14.3/kioslave/smb/kio_smb_auth.cpp b/kdebase-kioslaves-4.14.3/kioslave/smb/kio_smb_auth.cpp
index 4d236b4..5a5ef28 100644
--- a/kdebase-kioslaves-4.14.3/kioslave/smb/kio_smb_auth.cpp
+++ b/kdebase-kioslaves-4.14.3/kioslave/smb/kio_smb_auth.cpp
@@ -188,6 +188,11 @@ bool SMBSlave::auth_initialize_smbc()
 	    return false;
 	}
 
+	if (smb_force_protocol && (smb_min_proto || smb_max_proto)) {
+		smbc_setOptionProtocols(smb_context, smb_min_proto, smb_max_proto);
+		kDebug(KIO_SMB) << "Executed smbc_setOptionProtocols";
+	}
+
 #ifdef DEPRECATED_SMBC_INTERFACE // defined by libsmbclient.h of Samba 3.2
 
 	/* New libsmbclient interface of Samba 3.2 */
diff --git a/kdebase-kioslaves-4.14.3/kioslave/smb/kio_smb_browse.cpp b/kdebase-kioslaves-4.14.3/kioslave/smb/kio_smb_browse.cpp
index 5253be9..4cfa9cf 100644
--- a/kdebase-kioslaves-4.14.3/kioslave/smb/kio_smb_browse.cpp
+++ b/kdebase-kioslaves-4.14.3/kioslave/smb/kio_smb_browse.cpp
@@ -175,7 +175,18 @@ KUrl SMBSlave::checkURL(const KUrl& kurl) const
     QString surl = kurl.url();
     if (surl.startsWith(QLatin1String("smb:/"))) {
         if (surl.length() == 5) // just the above
+	{
+            // Samba 4.8 changed default 'client max protocol' from NT1 to SMB3,
+            // together with fixes of Badlock, it broke browsing smb:// with
+            // default settings, because only NT1 (SMB1) protocol can browse network.
+            // Samba 4.10 made it possible to set min/max protocol version
+            // via libsmbclient API.
+            kDebug(KIO_SMB) << "Setting client max protocol = NT1 for browsing SMB network";
+            smb_force_protocol = 1;
+            smb_min_proto = "NT1";
+            smb_max_proto = "NT1";
             return kurl; // unchanged
+	}
 
         if (surl.at(5) != '/') {
             surl = "smb://" + surl.mid(5);
