--- a/startkde.cmake
+++ b/startkde.cmake
@@ -329,6 +329,11 @@
 
 XDG_CURRENT_DESKTOP=KDE
 export XDG_CURRENT_DESKTOP
+
+# At this point all the environment is ready, let's send it to kwalletd if running
+if test -n "PAM_KWALLET_LOGIN" ; then
+    env | socat STDIN UNIX-CONNECT:$PAM_KWALLET_LOGIN
+fi
 
 # We set LD_BIND_NOW to increase the efficiency of kdeinit.
 # kdeinit unsets this variable before loading applications.

