diff -ru a/generators/smoke/writeClasses.cpp b/generators/smoke/writeClasses.cpp
--- a/generators/smoke/writeClasses.cpp 2021-11-28 19:03:22.869000000 +0300
+++ b/generators/smoke/writeClasses.cpp 2021-11-28 19:04:16.552000000 +0300
@@ -469,7 +469,11 @@
     }

     // destructor
-    // if the class can't be instanstiated, a callback when it's deleted is unnecessary
+
+    // if the class can't be instanciated, a callback when it's
+    // deleted is unnecessary, but we still generate a private
+    // destructor declaration, because otherwise gcc (under recent
+    // standards) will unsuccessfully attempt to generate a destructor
     if (Util::canClassBeInstanciated(klass)) {
         out << "    ~" << smokeClassName << "() ";
         if (destructor && destructor->hasExceptionSpec()) {
@@ -481,6 +485,8 @@
             out << ") ";
         }
         out << QString("{ this->_binding->deleted(%1, (void*)this); }\n").arg(m_smokeData->classIndex[className]);
+    } else {
+        out << "private:\n    ~" << smokeClassName << "();\n";
     }
     out << "};\n";


